### PR TITLE
Fix compilation with qt < 5.2

### LINKED ISSUE #1671: compilation error in qt/masternodelist.cpp ERROR: ‘class QDateTime’ has no member named ‘offsetFromUtc’
hi there. I was trying to compile dash and the make command stopped here:

> 
>  CXX      qt/qt_libbitcoinqt_a-masternodelist.o
> qt/masternodelist.cpp: In member function ‘void MasternodeList::updateMyMasternodeInfo(QString, QString, masternode_info_t&)’:
> qt/masternodelist.cpp:193:172: **error: ‘class QDateTime’ has no member named ‘offsetFromUtc’**
>  Mn.fInfoValid ? infoMn.nTimeLastPing + QDateTime::currentDateTime().offsetFromUtc() : 0)));
>                                                                      ^~~~~~~~~~~~~
> qt/masternodelist.cpp: In member function ‘void MasternodeList::updateNodeList()’:
> qt/masternodelist.cpp:276:173: error: ‘class QDateTime’ has no member named ‘offsetFromUtc’
>  %Y-%m-%d %H:%M", mn.lastPing.sigTime + QDateTime::currentDateTime().offsetFromUtc())));
>                                                                      ^~~~~~~~~~~~~
> At global scope:
> cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
> Makefile:6850: recipe for target 'qt/qt_libbitcoinqt_a-masternodelist.o' failed
> make[2]: *** [qt/qt_libbitcoinqt_a-masternodelist.o] Error 1
> make[2]: Leaving directory '/home/d/src3/dash/src'
> Makefile:9045: recipe for target 'all-recursive' failed
> make[1]: *** [all-recursive] Error 1
> make[1]: Leaving directory '/home/d/src3/dash/src'
> Makefile:652: recipe for target 'all-recursive' failed
> make: *** [all-recursive] Error 1

Does anyone knows whats wrong?
I am using Qt version 4.8.7 in /usr/lib/i386-linux-gnu

### PR DESCRIPTION
**Pull request by UdjinM6**
Should fix #1671

### CODE REVIEW

**Comment by codablock** [comment]
utACK

File: src/qt/masternodelist.cpp [hunk 1]
```diff
@@ -263,6 +270,13 @@ void MasternodeList::updateNodeList()
     ui->tableWidgetMasternodes->clearContents();
     ui->tableWidgetMasternodes->setRowCount(0);
     std::map<COutPoint, CMasternode> mapMasternodes = mnodeman.GetFullMasternodeMap();
+#if QT_VERSION < 0x050200
+    const QDateTime dateTime1 = QDateTime::currentDateTime();
+    const QDateTime dateTime2 = QDateTime(dateTime1.date(), dateTime1.time(), Qt::UTC);
+    int offsetFromUtc = dateTime1.secsTo(dateTime2);
+#else
+    int offsetFromUtc = QDateTime::currentDateTime().offsetFromUtc();
+#endif
```
Comment (gladcow) [inline src/qt/masternodelist.cpp]: May be it is better to move this code to separate local static function instead of copy-pasting it in 2 places?

Comment (schinzelh) [reply src/qt/masternodelist.cpp]: Yes please, no code duplication if possible

**Comment by thelazier** [comment]
Is `offsetFromUtc` equal to `GetTime()` ?

**Comment by UdjinM6** [comment]
>Is offsetFromUtc equal to GetTime() ?

no...
`offset from utc` == `local time` - `utc time`

File: src/qt/masternodelist.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/qt/masternodelist.cpp]: fixed

**Comment by gladcow** [comment]
utACK

**Pull request merged**